bundles/org.openhab.binding.miio/src/main/java/org/openhab/binding/miio/internal/basic/MiIoBasicChannel.java

```diff
@@ -25,102 +28,111 @@
+    private @Nullable String transfortmation;
     @SerializedName("ChannelGroup")
     @Expose
-    private String channelGroup;
+    private @Nullable String channelGroup;
     @SerializedName("actions")
     @Expose
-    private List<MiIoDeviceAction> miIoDeviceActions = new ArrayList<MiIoDeviceAction>();
+    private @Nullable List<MiIoDeviceAction> miIoDeviceActions = new ArrayList<MiIoDeviceAction>();
 
     public String getProperty() {
-        return property;
+        final String property = this.property;
+        return (property != null) ? property : "";
     }
 
     public void setProperty(String property) {
         this.property = property;
     }
 
     public String getFriendlyName() {
-        return type == null || friendlyName.isEmpty() ? channel : friendlyName;
+        final String fn = friendlyName;
+        return (fn == null || type == null || fn.isEmpty()) ? getChannel() : fn;
     }
 
     public void setFriendlyName(String friendlyName) {
         this.friendlyName = friendlyName;
     }
 
     public String getChannel() {
-        return channel;
+        final @Nullable String channel = this.channel;
+        return channel != null ? channel : "";
     }
 
     public void setChannel(String channel) {
         this.channel = channel;
     }
 
     public String getChannelType() {
-        return channelType == null || channelType.isEmpty() ? BINDING_ID + ":" + channel
-                : (channelType.startsWith("system") ? channelType : BINDING_ID + ":" + channelType);
+        final @Nullable String ct = channelType;
+        return ct == null || ct.isEmpty() ? BINDING_ID + ":" + channel
+                : (ct.startsWith("system") ? ct : BINDING_ID + ":" + ct);
     }
 
     public void setChannelType(String channelType) {
         this.channelType = channelType;
     }
 
     public String getType() {
-        return type == null ? "" : type;
+        final @Nullable String type = this.type;
+        return type != null ? type : "";
     }
 
     public void setType(String type) {
         this.type = type;
     }
 
     public Boolean getRefresh() {
-        return refresh && !property.isEmpty();
+        final @Nullable Boolean rf = refresh;
+        return rf != null && rf.booleanValue() && !getProperty().isEmpty();
     }
 
     public void setRefresh(Boolean refresh) {
         this.refresh = refresh;
     }
 
     public String getChannelGroup() {
-        return channelGroup;
+        final @Nullable String channelGroup = this.channelGroup;
+        return channelGroup != null ? channelGroup : "";
     }
 
     public void setChannelGroup(String channelGroup) {
         this.channelGroup = channelGroup;
     }
 
     public List<MiIoDeviceAction> getActions() {
-        return miIoDeviceActions;
+        final @Nullable List<MiIoDeviceAction> miIoDeviceActions = this.miIoDeviceActions;
+        return (miIoDeviceActions != null) ? miIoDeviceActions : new ArrayList<MiIoDeviceAction>();
```

Comment:
If these aren't supposed to be mutable you should return a Collections.emptyList() instead.

Reply:
thanks.. changed it.